distributed/protocol/serialize.py

```diff
@@ -772,6 +772,8 @@ def _deserialize_array(header, frames):
 def _serialize_memoryview(obj):
     if obj.format == "O":
         raise ValueError("Cannot serialize `memoryview` containing Python objects")
+    if not obj and obj.ndim > 1:
+        raise ValueError("Cannot serialize empty non-1-D `memoryview`")
```

Comment:
Since we can't properly reconstruct these during deserialization, just raise here. This should be as close as possible to user code. So should hopefully make it clearer what data triggered this error.